x-pack/plugins/enterprise_search/common/types/connectors.ts

```diff
@@ -60,6 +60,7 @@ export type ConnectorConfiguration = Record<
   string,
   ConnectorConfigProperties | ConnectorConfigCategoryProperties | null
 > & {
+  document_level_security?: ConnectorConfigProperties;
```

Comment:
```suggestion
  use_document_level_security?: ConnectorConfigProperties;
```

I've used `use_document_level_security` in the python backend to be consistent with `use_text_extraction_service`. Maybe we should use `use_document_level_security` here, too?

Reply:
This will require changes in multiple places. I will do them.